### PR TITLE
Discussion Only: Terraform Import

### PR DESCRIPTION
**Pull request by johnrengelman**
This is a simple POC for discussion purposes only. It is missing automated tests and none of the command switches have been tested.

Syntax is:
```bash
$ terraform import <resource_name> <resource_id>
```

Example:
```bash
$ terraform import aws_vpc.main vpc-123456
```

### CODE REVIEW

**Comment by johnrengelman** [comment]
This seems to work for the most part. I have noticed that some resources don't set all attributes during a `read`, so even after importing there are usually differences in the plan.

So to handle that, we could do the following:
* Live with it and fix the various resources that are dropping the ball as they are reported.
* Add an additional `import-attribute` command that directly sets an attribute to the supplied value for a resource
* Somehow tie into the Resource schema and ensure that any "default" values exist in the state file during a 'refresh'.

Also, right now this command requires a `terraform refresh` after running `terraform import` to actually populate the state file. We should integrate this so the `import` executes a `refresh` on the resource immediately after importing.

File: command/import.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	return 0
+}
+
+func (c *ImportCommand) Help() string {
+	helpText := `
+Usage: terraform import [options] name id
+
+  Manually mark a resource as tainted, forcing a destroy and recreate
+  on the next plan/apply.
+
```
Comment (blalor) [inline command/import.go]: This description looks incorrect; copied from `taint`?

**Comment by blalor** [comment]
What about generating the config for the `.tf` file?  If I have a resource in AWS that I want to import into my Terraform config, I still don't want to have to deal with copying the attributes in by hand.

**Comment by johnrengelman** [comment]
Baby steps. That would be a bit harder to do I think.

**Comment by blalor** [comment]
I've done a little test with this and it works well.  I had to rebase it onto the current master because the deps have moved on and they're neither vendored nor managed. :bomb: Anyway, this is very promising!

**Comment by AlexanderEkdahl** [comment]
This is awesome! :snowman: 

I don't think it should automatically run refresh as there could be multiple resources that should be imported before running a refresh.

**Comment by johnrengelman** [comment]
Updated this PR with some doc fixes.

I'm not sure about the automatically running a refresh or not. I was thinking that we could run a targeted refresh only for the resources imported but I got lost in the command code a bit trying to figure that out. 

I think extending this concept to allow for the following would be beneficial but I'm not sure about how to do it:
* import multiple resources in a single invocation
* import resources from a file

The first option would look like: `terraform import --resource aws.vpc.main=vpc-123456 --resource aws.vpc.dev=vpc-987654` (not sure about the syntax)

The second option would be something like: `terraform import --file resources.properties`
where `resource.properties` looked like:
```
aws.vpc.main=vpc-123456
aws.vpc.dev=vpc-987654
```

**Comment by johnrengelman** [comment]
Any feedback from the core team on this one? @mitchellh @catsby?

**Comment by mitchellh** [comment]
I'll review this shortly, thanks for the patience @johnrengelman

**Comment by josephholsten** [comment]
@mitchellh had a chance to review? I've got folks doing this sort of thing manually now, this little cli sugar and validation would go a long way.

**Comment by ross** [comment]
Messed around with an alternate approach over the weekend that instead adds a parameter to the `refresh` command. https://github.com/hashicorp/terraform/pull/3345. It addresses something I ran in to when trying to import my personal setup using this branch. Namely I wanted to generate a set of mappings (done by hand for now) between resources and their AWS ids and iterate on the .tf files to match those existing resources. It ignores things that are already managed as well as things that aren't yet configured. 

It's strongly based on this PR, really started with the code here and dug in to the other pieces I needed to touch. Thoughts & suggestions welcome.

**Comment by josephholsten** [comment]
@phinze this needs :eyes:, is this worth pursuing? needs tests & doc, but promising.

**Comment by stack72** [comment]
Closed as we have Terraform Import working in 0.7 :)
